Key resolution options: require x509_hash, remove web based key resolution (#178)
openid/oid4vc-haip: openid4vc-high-assurance-interoperability-profile-1_0.md

* Response type MUST be `vp_token`.
* Response mode MUST be `direct_post.jwt`. The Verifier MUST return `redirect_uri` in response to the HTTP POST request from the Wallet, where the Wallet redirects the User to, as defined in Section 8.2 of [@!OIDF.OID4VP]. Implementation considerations for the response mode `direct_post.jwt` are given in Section 14.3 of [@!OIDF.OID4VP].
* Authorization Request MUST be sent using the `request_uri` parameter as defined in JWT-Secured Authorization Request (JAR) [@!RFC9101].

> * For Client Identifier Schemes as introduced in Section 5.10 of [@!OIDF.OID4VP], `x509_hash` MUST be supported

Reviewer: ```suggestion
* For Client Identifier Schemes as introduced in Section 5.10 of [@!OIDF.OID4VP], `x509_hash` MUST be supported.
```

> * For Client Identifier Schemes as introduced in Section 5.10 of [@!OIDF.OID4VP], `x509_hash` MUST be supported.

Reviewer 2: Section number has changed. I'm also not sure on what is actually meant, does this match the intention:

```suggestion
* For Client Identifier Schemes as introduced in Section 5.9 of [@!OIDF.OID4VP], `x509_hash` MUST be supported by the Wallet and used by the Verifier.
```

* The `iss` claim MUST be an HTTPS URL. The `iss` value is used to obtain Issuer’s signing key as defined in (#issuer-key-resolution).
* The `vct` JWT claim as defined in [@!I-D.ietf-oauth-sd-jwt-vc].
* The `cnf` claim [@!RFC7800] MUST conform to the definition given in [@!I-D.ietf-oauth-sd-jwt-vc]. Implementations conforming to this profile MUST include the JSON Web Key [@!RFC7517] in the `jwk` sub claim.

> * The public key used to validate the signature on the Status List Token MUST be included in the x5c JOSE header of the Token.

Reviewer: ```suggestion
* The public key used to validate the signature on the Status List Token MUST be included in the `x5c` JOSE header of the Token.
```

## Issuer identification and key resolution to validate an issued Credential {#issuer-key-resolution}

> This profile supports one mandatory way to represent and resolve the key required to validate the issuer signature of an SD-JWT VC, x.509 certificates. This MUST be supported by all entities (Issuer, Wallet, Verifier).

Reviewer: ```suggestion
This profile supports one mandatory way to represent and resolve the key required to validate the issuer signature of an SD-JWT VC, x.509 certificates. This MUST be supported by all entities (Issuer, Wallet, Verifier):
```

> * x.509 certificates: the SD-JWT VC contains the issuer's certificate along with a trust chain in the `x5c` JOSE header. In this case, the `iss` value MUST be an URL with a FQDN matching a `dNSName` Subject Alternative Name (SAN) [@!RFC5280] entry in the leaf certificate.

Reviewer 2: I think we need to see something about 'MUST validate against a root certificate but how you get the root certificate is out of scope' ?

Support for web-based key resolution is RECOMMENDED:

> * Web-based key resolution: The key used to validate the Issuer’s signature on the SD-JWT VC MUST be obtained from the SD-JWT VC issuer's metadata as defined in Section 5 of [@!I-D.ietf-oauth-sd-jwt-vc]. The JOSE header `kid` MUST be used to identify the respective key.

Reviewer 2: I'm not 100% sure I understand how this works. It seems to conflict with the requirement to support x5c. 

I suspect the text referred to has been moved to section 3.5 ( https://www.ietf.org/archive/id/draft-ietf-oauth-sd-jwt-vc-08.html#section-3.5  ) but also I think it's been renamed or something as I'm not clear which method in that section needs to be used.

Reviewer 3: I'm also not sure how this works, and if it's required to support X5C why support this other option, since you can't rely on it? Also since the Status List MUST use `x5c`, you then get like this weird combination of using web based key resolution vs X509 based.

## Issuer identification and key resolution to validate an issued Credential {#issuer-key-resolution}

> This profile supports one mandatory way to represent and resolve the key required to validate the issuer signature of an SD-JWT VC, x.509 certificates. This MUST be supported by all entities (Issuer, Wallet, Verifier):

Reviewer 4: > This MUST be supported by all entities (Issuer, Wallet, Verifier):
Does this mean that existing issuers utilizing web-based key resolution are no longer HAIP compatible? I disgree that web-based key resolution should be considered HAIP incompatible.

Author: My current interpretation would be: Every entity has to support the x.509 based key resolution / trust establishment for sd-jwt vc to be HAIP compatible. The different entities might also support other ways like web-based key resolution, but for interop we have to mandate support for at least one mechanism everyone implements.

> This profile supports one mandatory way to represent and resolve the key required to validate the issuer signature of an SD-JWT VC, x.509 certificates. This MUST be supported by all entities (Issuer, Wallet, Verifier):

Author: Would this be a better wording?
```suggestion
This profile supports one way that is mandatory to support to represent and resolve the key required to validate the issuer signature of an SD-JWT VC, x.509 certificates. This MUST be supported by all entities (Issuer, Wallet, Verifier):
```

Reviewer 2: Maybe this?

```suggestion
This profile mandates the support for x.509 certifications to represent and resolve the key required to validate the issuer signature of an SD-JWT VC. This MUST be supported by all entities (Issuer, Wallet, Verifier):
```

But I think we actually need to combine this paragraph and the single bullet into a single hopefully simpler paragraph.

> * x.509 certificates: the SD-JWT VC contains the issuer's certificate along with a trust chain in the `x5c` JOSE header. In this case, the `iss` value MUST be an URL with a FQDN matching a `dNSName` Subject Alternative Name (SAN) [@!RFC5280] entry in the leaf certificate.

Reviewer 5: As of the recent -10 SD-JWT VC draft, the issuer claim is no longer used in the X.509 Certificate-based Issuer Signature Mechanism  https://author-tools.ietf.org/iddiff?url1=draft-ietf-oauth-sd-jwt-vc-09&url2=draft-ietf-oauth-sd-jwt-vc-10&difftype=--html 
```suggestion
* x.509 certificates: the SD-JWT VC contains the issuer's certificate along with a trust chain in the `x5c` JOSE header.
```

## Issuer identification and key resolution to validate an issued Credential {#issuer-key-resolution}

> This profile mandates the support for x.509 certificates key resolution to validate the issuer signature of an SD-JWT VC. This MUST be supported by all entities (Issuer, Wallet, Verifier). The SD-JWT VC MUST contain the issuer's certificate along with a trust chain in the `x5c` JOSE header parameter as described in section 3.5 of [@!I-D.ietf-oauth-sd-jwt-vc].

Reviewer 5: ```suggestion
This profile mandates the support for X.509 certificate-based key resolution to validate the issuer signature of an SD-JWT VC. This MUST be supported by all entities (Issuer, Wallet, Verifier). The SD-JWT VC MUST contain the credential issuer's certificate along with a trust chain in the `x5c` JOSE header parameter as described in section 3.5 of [@!I-D.ietf-oauth-sd-jwt-vc].
```

   -04

>    * Define key resolution mechanisms for all elements

Reviewer 2: I think this underplays "x509_san_dns & verifier_attestations client id prefixes are no longer permitted, x509_hash must be used"

Reviewer: ```suggestion
   * x509_san_dns & verifier_attestations client id prefixes are no longer permitted, x509_hash must be used
```

   -04

>    * x509_san_dns & verifier_attestations client id prefixes are no longer permitted, x509_hash must be used

Reviewer 2: ```suggestion
   * x509_san_dns & verifier_attestations client id prefixes are no longer permitted, x509_hash must be used
   * x.509 certificates are now the mandatory mechanism for SD-JWT VC issuer key resolution
   * `x5c` header in Status List Token must be present
```
